File: 003FizzBuzz/Java/FizzBuzz.java

public class FizzBuzz {

    static private int FirstValue = 1;
    static private int LastValue = 100;

    public static void main(String[] args){
        fizzbuzz();
    }

    public static void fizzbuzz(){
        for(int i=FirstValue; i<=LastValue; i++){
            if(i%15==0){
                System.out.println("FizzBuzz");
            }
            else if(i%3==0){
                System.out.println("Fizz");
            }
            else if(i%5==0){
                System.out.println("Buzz");
            }
            else{
                System.out.println(i);
            }
        }
    }
}

File: 003FizzBuzz/Scala/FizzBuzz.scala



object FizzBuzz extends App {
    
    val first_value = 1
    val last_value = 100
    
    fizzbuzz(first_value, last_value)

    def fizzbuzz(num:Int, last:Int){
        if(num%15 == 0){
            println("FizzBuzz")
        }
        else if(num%3 == 0){
            println("Fizz")
        }
        else if(num%5 == 0){
            println("Buzz")
        }
        else{
            println(num)
        }
        
        if(num < last){
            fizzbuzz(num+1, last)
        }
    }
    
}


